pkg/controllers/cloud_config_sync_controller_test.go

```diff
@@ -217,6 +233,14 @@ var _ = Describe("Cloud config sync controller", func() {
 		Eventually(
 			apierrors.IsNotFound(cl.Get(ctx, client.ObjectKeyFromObject(infra), infra)),
 		).Should(BeTrue())
+
+		networkResource := makeNetworkResource()
+		Expect(cl.Delete(ctx, networkResource)).To(Succeed())
+
+		Eventually(func() bool {
+			err := cl.Get(ctx, client.ObjectKeyFromObject(networkResource), networkResource)
+			return apierrors.IsNotFound(err)
+		}).Should(BeTrue())
```

Comment:
True/False assertions can be hard to understand when they fail, if you want to do this, add context to the assertion
```suggestion
		}).Should(BeTrue(), "Expected not found error")
```